### DISCUSSION TITLE
Prewiew should not be rendered in unrelated pull requests

### DISCUSSION THREAD

**Discussion by Czaki** [chanzuckerberg/napari-hub#389] [Ideas]
I have installed napari-hub-bot in the PartSeg repository and it adds a nice preview of the napari hub page, but it happens for all pull requests, not only this one which touch description:

https://github.com/4DNucleome/PartSeg/pull/504#issuecomment-1013229837

I think, that is not an optimal solution. It should be easy to limit it only to PR which interacts with DESCRIPTION.md/Readme, etc.

**Comment by neuromusic** [comment]
I can see how this would be annoying @Czaki :D 

only triggering either the action or the bot when key assets are touched is certainly one approach, however, in the future, we may start adding metadata to the hub which has more complex dependencies (such as testing that the plugin installs into the latest napari)

I could also imaging a few other solutions, such as...

- making the preview page "on demand" rather than automated by letting the plugin developer tag the bot in a comment (e.g. "@napari-hub-bot preview")
- the napari hub app could cache the preview HTML and only post the preview comment if there's a change in what is rendered

**Reply by codemonkey800** [reply]
The initial prototype required adding the label `napari-hub-preview`to start the preview build process. Another thing we could do is detect specific files like `setup.py` or `.napari/`, and add support for a custom list of file patterns to trigger the build on.

**Comment by neuromusic** [comment]
I'm curious which is most annoying here: the action that builds on every PR or the comment from the bot? (or both 🤣 )

**Reply by codemonkey800** [reply]
In the prototype, I also had it set up to update an existing comment. I also know that the cloudflare bot does something similar, so might be something we could do lol
